datafusion/core/src/physical_plan/aggregates/mod.rs

```diff
@@ -270,6 +267,18 @@ pub struct AggregateExec {
     required_input_ordering: Option<LexOrderingReq>,
 }
 
+/// Check whether aggregations is in first stage.
+/// It means that, aggregation input is not another aggregation
+/// `merge_batch` method will not be called for these modes.
+fn is_aggregation_first_stage(mode: &AggregateMode) -> bool {
```

Comment:
Maybe this would be easier to find if it were a method on `AggregateMode`

```rust
impl AggregateMode {
... 
 pub fn is_first_stage(&self) -> bool {....}
}
```

Reply:
moved implementation to the method.